File: sumaup360-backend/src/main/java/com/sumaup360/app/peya/domain/PeyaUpload.java
package com.sumaup360.app.peya.domain;

import com.sumaup360.app.peya.enums.PeyaStatus;
import com.sumaup360.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.util.UUID;

/** Carga mensual de PDF de ventas Peya y su procesamiento. */
@Entity
@Table(name = "peya_upload", schema = "app")
@Getter
@Setter
public class PeyaUpload extends BaseEntity {

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "ruc", length = 11)
    private String ruc;

    @Column(name = "periodo", nullable = false, length = 7)
    private String periodo; // AAAA-MM

    @Column(name = "pdf_url", columnDefinition = "text", nullable = false)
    private String pdfUrl;

    @Enumerated(EnumType.STRING)
    @Column(name = "estado", nullable = false, length = 30)
    private PeyaStatus estado = PeyaStatus.PDF_SUBIDO;

    @Column(name = "observacion_usuario", length = 500)
    private String observacionUsuario;

    @Column(name = "observacion_backoffice", length = 500)
    private String observacionBackoffice;

    @Column(name = "codigo_nps", length = 40)
    private String codigoNps;

    @Column(name = "completed_at")
    private OffsetDateTime completedAt;
}

File: sumaup360-mobile/sumaup360_app/lib/features/peya/domain/peya_upload.dart
// Carga mensual de Peya (cara repartidor) + archivos resultado.

class PeyaUpload {
  const PeyaUpload({
    required this.id,
    this.periodo,
    this.pdfUrl,
    this.estado,
    this.observacionUsuario,
    this.observacionBackoffice,
    this.codigoNps,
    this.createdAt,
    this.completedAt,
  });

  final String id;
  final String? periodo;
  final String? pdfUrl;
  final String? estado;
  final String? observacionUsuario;
  final String? observacionBackoffice;
  final String? codigoNps;
  final String? createdAt;
  final String? completedAt;

  factory PeyaUpload.fromJson(Map<String, dynamic> j) => PeyaUpload(
        id: j['id'] as String,
        periodo: j['periodo'] as String?,
        pdfUrl: j['pdfUrl'] as String?,
        estado: j['estado'] as String?,
        observacionUsuario: j['observacionUsuario'] as String?,
        observacionBackoffice: j['observacionBackoffice'] as String?,
        codigoNps: j['codigoNps'] as String?,
        createdAt: j['createdAt'] as String?,
        completedAt: j['completedAt'] as String?,
      );
}

class PeyaFile {
  const PeyaFile({required this.id, required this.fileUrl, this.fileName});
  final String id;
  final String fileUrl;
  final String? fileName;

  factory PeyaFile.fromJson(Map<String, dynamic> j) => PeyaFile(
        id: j['id'] as String,
        fileUrl: (j['fileUrl'] as String?) ?? '',
        fileName: j['fileName'] as String?,
      );
}
